src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecuritySessionHistory.jsx

```diff
@@ -0,0 +1,98 @@
+import {Flex, Box} from 'grid-emotion';
+import {withRouter} from 'react-router';
+import React from 'react';
+
+import {t} from 'app/locale';
+import AsyncView from 'app/views/asyncView';
+import ListLink from 'app/components/listLink';
+import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
+import {Panel, PanelBody, PanelHeader, PanelItem} from 'app/components/panels';
+import TimeSince from 'app/components/timeSince';
+import recreateRoute from 'app/utils/recreateRoute';
+
+class SessionRow extends React.Component {
+    static propTypes = {
+      ipAddress: PropTypes.string.isRequired,
+      countryCode: PropTypes.string,
+      regionCode: PropTypes.string,
+      lastSeen: PropTypes.string.isRequired,
+      firstSeen: PropTypes.string.isRequired,
+    };
+
+    render() {
+      let {ipAddress, countryCode, regionCode, lastSeen, firstSeen} = this.props;
+
+      return (
+        <PanelItem justify="space-between">
+          <Flex align="center" flex={1}>
+            <Box flex="1">
+            <div style={{marginBottom: 5}}><strong>{ipAddress}</strong></div>
+            {countryCode && regionCode &&
+                <div><small>{countryCode} ({regionCode})</small></div>
+            }
+            </Box>
+          </Flex>
+          <Flex align="center" w={140} mx={2}>
+            <small><TimeSince date={firstSeen} /></small>
+          </Flex>
+          <Flex align="center" w={140} mx={2}>
+          <small><TimeSince date={lastSeen} /></small>
+          </Flex>
+        </PanelItem>
+      );
+    }
+  }
+
+class AccountSecuritySessionHistory extends AsyncView {
+  getTitle() {
+    return t('Session History');
+  }
+
+  getEndpoints() {
+    return [['ipList', '/users/me/ips/']];
+  }
+
+  renderBody() {
+    let {ipList} = this.state;
+
+    return (
+      <div>
+        <SettingsPageHeader title="Security" tabs={
+            <ul className="nav nav-tabs" style={{borderBottom: '1px solid #ddd'}}>
```

Comment:
Should this be a styled component? It shows up a few times in this pull request.

Reply:
:+1: sounds good.